harmony-kotlin/src/commonTest/kotlin/com/harmony/kotlin/data/repository/CacheRepositoryTests.kt

```diff
@@ -0,0 +1,463 @@
+@file:Suppress("IllegalIdentifier")
+
+package com.harmony.kotlin.data.repository
+
+import com.harmony.kotlin.common.randomString
+import com.harmony.kotlin.common.runTest
+import com.harmony.kotlin.data.datasource.memory.InMemoryDataSource
+import com.harmony.kotlin.data.datasource.memory.anyInMemoryDataSource
+import com.harmony.kotlin.data.error.DataNotFoundException
+import com.harmony.kotlin.data.error.ObjectNotValidException
+import com.harmony.kotlin.data.error.OperationNotAllowedException
+import com.harmony.kotlin.data.operation.CacheOperation
+import com.harmony.kotlin.data.operation.CacheSyncOperation
+import com.harmony.kotlin.data.operation.DefaultOperation
+import com.harmony.kotlin.data.operation.MainOperation
+import com.harmony.kotlin.data.operation.MainSyncOperation
+import com.harmony.kotlin.data.operation.anyOperation
+import com.harmony.kotlin.data.query.anyKeyQuery
+import com.harmony.kotlin.data.query.anyQuery
+import com.harmony.kotlin.data.query.anyVoidQuery
+import com.harmony.kotlin.data.utilities.InsertionValue
+import com.harmony.kotlin.data.utilities.InsertionValues
+import com.harmony.kotlin.data.utilities.anyInsertionValue
+import com.harmony.kotlin.data.utilities.anyInsertionValues
+import com.harmony.kotlin.data.validator.Validator
+import com.harmony.kotlin.data.validator.anyMockValidator
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlin.test.Test
+import kotlin.test.assertContentEquals
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+
+@ExperimentalCoroutinesApi
+class CacheRepositoryTests {
+
+  //region get() - tests
+
+  @Test
+  fun `should retrieves the value from the main datasource when MainOperation is provided when get() is called`() = runTest {
```

Comment:
Picky suggestion, no need to change it: To understand the test better I would avoid those two 'whens'

Maybe rephrase from this:
- should retrieves the value from the main datasource **when** MainOperation is provided **when** get() is called
to something like:
- get() should retrieve the value from the main datasource when MainOperation is provided

The same happens with the rest of the tests

Reply:
Some changes applied to #60